utils/azure/storage.py

```diff
@@ -1 +1,27 @@
-pass
+from azure.identity import DefaultAzureCredential
+from azure.storage.blob import BlobServiceClient
+
+
+def instantiate_blob_client(
+    sp_credential: DefaultAzureCredential | None = None, account_url: str | None = None
+) -> BlobServiceClient:
+    """Function to instantiate blob storage client to interact
+    with Azure Storage.
+    Args:
+        sp_credential (DefaultAzureCredential): Service principal credential object
+        for use in authenticating with Storage API.
+        account_url (str): URL of the storage account.
+    Returns:
+        BlobServiceClient: Azure Blob Storage client
+    Raises:
+        ValueError: If sp_credential is invalid or BlobServiceClient
+        fails to instantiate.
+    """
+
+    if not sp_credential:
+        raise ValueError("Service principal credential not provided.")
```

Comment:
```suggestion
    if not sp_credential:
        raise ValueError("Service principal credential not provided.")

    if not account_url:
        raise ValueError("Account URL not provided.")
```
I think we should either check that it's not `None`, or make it `str` only